pkg/controllers/provisioning/v1alpha1/reallocation/terminate.go

```diff
@@ -60,18 +61,16 @@ func (t *Terminator) cordonNodes(ctx context.Context, provisioner *v1alpha1.Prov
 	}
 	// 2. Cordon nodes
 	for _, node := range nodeList {
-		if !node.Spec.Unschedulable {
-			persisted := node.DeepCopy()
-			node.Spec.Unschedulable = true
-			node.Labels = functional.UnionStringMaps(
-				node.Labels,
-				map[string]string{v1alpha1.ProvisionerPhaseLabel: v1alpha1.ProvisionerDrainingPhase},
-			)
-			if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-				return fmt.Errorf("patching node %s, %w", node.Name, err)
-			}
-			zap.S().Debugf("Cordoned node %s", node.Name)
+		persisted := node.DeepCopy()
+		node.Spec.Unschedulable = true
+		node.Labels = functional.UnionStringMaps(
+			node.Labels,
+			map[string]string{v1alpha1.ProvisionerPhaseLabel: v1alpha1.ProvisionerDrainingPhase},
+		)
+		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
+			return fmt.Errorf("patching node %s, %w", node.Name, err)
 		}
+		zap.S().Debugf("Cordoned node %s", node.Name)
```

Comment:
Is it possible that this logs too frequently?

Reply:
I don't think so. I think the log makes sense as a debug message and is only logging when necessary.  After a node is labeled with the `v1alpha1.ProvisionerPhaseLabel` it won't be cordoned again. The cordon should only be logged once per node that is cordoned.

Reply:
This should maybe actually be an Info? Is there going to be any other log line indicating that the node was cordoned by Karpenter?

Reply:
It could get unnecessarily noisy in a large cluster. wrt to investigating who cordoned the node though, that should be fairly easy with the label that's added when cordoning nodes. In that situation, it really only matters if karpenter cordoned the node but for some reason didn't terminate it. If it's terminated, the log is at info level anyways so I don't think it's a strong reason to move this log up to info.

Reply:
Agree. I want to be really stingy with Info logs. As we get debugging feedback, we can think about promoting some debug logs to info. We discussed this a month or so back and decided that top level aggregate steps (e.g. "cordoned 5 nodes") would be info level, while individual workflow steps (e.g. "cordoned node foo") would be debug.

Reply:
That's fine - as long as there's a way to figure out what entity did the cordoning.